testing/gomockmatchers/booltype.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package gomockmatcher
+
+import (
+	"fmt"
+
+	"github.com/golang/mock/gomock"
+)
+
+type BoolType struct{ b bool }
+
+func OfTypeBool(b bool) gomock.Matcher {
+	return &BoolType{b}
+}
+
+func (o *BoolType) Matches(x interface{}) bool {
+	return o.b == x
```

Comment:
should this be 

```go
func (o *BoolType) Matches(x interface{}) bool {
	if b, ok := x.(bool) {
		return o.b == b
	}
	return false
}
```

or

```go
func (o *BoolType) Matches(x interface{}) {
	return o.b == x.(bool)
}
```

Reply:
@SimonRichardson thx making the change - i'm going to put the test changes in a different pr, so i can get the other stuff landed.